File: Java-Postgres-CRUD-main/src/model/Employer.java
package model;

import java.util.Date;

public class Employer extends User {
    private Date fechaContrato;
    private String departamento;
    private double salario;
    private String rango;
    private double descuento;
    private String IBAN;

    public Employer() {
        super();
        this.setTipoUsuario("EMPLOYER");
    }

    public Employer(int idUser, int numeroTelefono, String email, String nombre,
                    Date fechaContrato, String departamento, double salario,
                    String rango, double descuento, String IBAN) {
        super(idUser, numeroTelefono, email, nombre, "EMPLOYER");
        this.fechaContrato = fechaContrato;
        this.departamento = departamento;
        this.salario = salario;
        this.rango = rango;
        this.descuento = descuento;
        this.IBAN = IBAN;
    }

    public Date getFechaContrato() {
        return fechaContrato;
    }

    public void setFechaContrato(Date fechaContrato) {
        this.fechaContrato = fechaContrato;
    }

    public String getDepartamento() {
        return departamento;
    }

    public void setDepartamento(String departamento) {
        this.departamento = departamento;
    }

    public double getSalario() {
        return salario;
    }

    public void setSalario(double salario) {
        this.salario = salario;
    }

    public String getRango() {
        return rango;
    }

    public void setRango(String rango) {
        this.rango = rango;
    }

    public double getDescuento() {
        return descuento;
    }

    public void setDescuento(double descuento) {
        this.descuento = descuento;
    }

    public String getIBAN() {
        return IBAN;
    }

    public void setIBAN(String IBAN) {
        this.IBAN = IBAN;
    }

    @Override
    public String toString() {
        return "Employer{" +
                "idUser=" + getIdUser() +
                ", nombre='" + getNombre() + '\'' +
                ", email='" + getEmail() + '\'' +
                ", telefono=" + getNumeroTelefono() +
                ", fechaContrato=" + fechaContrato +
                ", departamento='" + departamento + '\'' +
                ", salario=" + salario +
                ", rango='" + rango + '\'' +
                ", descuento=" + descuento +
                ", IBAN='" + IBAN + '\'' +
                '}';
    }
}
 

File: proyecto/php/src/model/Employer.php
<?php
require_once "User.php";

class Employer extends User {
    private $fechaContrato;
    private $departamento;
    private $salario;
    private $rango;
    private $descuento;
    private $IBAN;

    public function __construct($idUser = null, $numeroTelefono = null, $email = null, $nombre = null,
                                $fechaContrato = null, $departamento = null, $salario = null,
                                $rango = null, $descuento = null, $IBAN = null) {
        parent::__construct($idUser, $numeroTelefono, $email, $nombre, "EMPLOYER");
        $this->fechaContrato = $fechaContrato;
        $this->departamento = $departamento;
        $this->salario = $salario;
        $this->rango = $rango;
        $this->descuento = $descuento;
        $this->IBAN = $IBAN;
    }

    // Getters y setters ...
    public function getFechaContrato() { return $this->fechaContrato; }
    public function getDepartamento() { return $this->departamento; }
    public function getSalario() { return $this->salario; }
    public function getRango() { return $this->rango; }
    public function getDescuento() { return $this->descuento; }
    public function getIBAN() { return $this->IBAN; }

    public function setFechaContrato($f) { $this->fechaContrato = $f; }
    public function setDepartamento($d) { $this->departamento = $d; }
    public function setSalario($s) { $this->salario = $s; }
    public function setRango($r) { $this->rango = $r; }
    public function setDescuento($d) { $this->descuento = $d; }
    public function setIBAN($i) { $this->IBAN = $i; }

    public function __toString() {
        return parent::__toString() .
               ", fechaContrato=" . $this->fechaContrato .
               ", departamento='" . $this->departamento . "'" .
               ", salario=" . $this->salario .
               ", rango='" . $this->rango . "'" .
               ", descuento=" . $this->descuento .
               ", IBAN='" . $this->IBAN . "'}";
    }
}
